### DISCUSSION TITLE
shared place with "MyCustomTags"

### DISCUSSION THREAD

**Discussion by HerzScheisse** [vesta-webtrees-2-custom-modules/vesta_shared_places#187] [Q&A] [answered]
Hey ric,

I've installed the MyCustomTags Module from Jefferson and added some tags that i want to use...
For quite a long time i have been using this via my modified webtrees installation but i nee to update this every time wt gets updated, so this is a new approach...

in the file MyCustomTags.php i've added:
```
use Fisharebest\Webtrees\Elements\AddressLine;
use Fisharebest\Webtrees\Elements\PlaceName;
...
    protected function customTags(): array
    {
        return [
            //Additonal custom tags
...
            'OBJE:PLAC'                     => new PlaceName(I18N::translate('Place of event')),
            'OBJE:PLAC:ADDR'                => new AddressLine(I18N::translate('Address')),
...
    protected function customSubTags(): array
    {
        return [
...
            'OBJE'       => [['PLAC', '0:1']],
            'OBJE:PLAC'  => [['_LOC', '0:1'], ['ADDR', '0:1'], ['_GOV', '0:1']],
```

so for the place everything is working fine... and also the autocomplete of place names works... 

Do you know how i can get the shared places and GOV Ids working

![mycustomtags_sharedplaces](https://github.com/user-attachments/assets/1bd974b4-cee1-41f4-9402-1231af214b37)

Thanks
/Holger

**Comment by ric2016** [comment, marked as answer]
You'll have to `registerTags` for the additional sub tags, and sub-sub tags ("OBJE:PLAC:_LOC" etc) as well, similar to the setup in [Shared Places](https://github.com/vesta-webtrees-2-custom-modules/vesta_shared_places/blob/84f2030d40d8333e8a732d52c39d6784dfc20078/SharedPlacesModule.php#L405) and [Gov4Webtrees](https://github.com/vesta-webtrees-2-custom-modules/vesta_gov4webtrees/blob/72e8c0202b70b686161ab9bfca7d95df9ddc9d2e/Gov4WebtreesModule.php#L148).

**Comment by HerzScheisse** [comment]
awesome... thank you very much... this was very helpfull and i got it working :)

**Comment by HerzScheisse** [comment]
one other question related to this...

If the media object is inside a shared place... the display of the shared place in media-list page works as expected...

But how can i achieve this also with images the other way arround... so the image is not part of a shared place... but has a shared place (hope this makes sense to you :-) )

![shared_place_issue_2](https://github.com/user-attachments/assets/e3efe478-2971-4d0d-8d4d-329a90531c0f)
![shared_place_issue_1](https://github.com/user-attachments/assets/a6b6d06c-73fa-4fb4-9ad0-fe0a127be76b)

**Comment by ric2016** [comment]
> the other way arround

The listed of linked objects is not supposed to work the other way around, it shows 'incoming' links. You'd have to extend the functionality in `ManageMediaData.mediaObjectInfo()`, or add the shared place further above, with the displayed fields (this would be conceptually less confusing) in `ManageMediaData.mediaFileInfo()`.

Both methods aren't implemented with extensibility in mind (should be, at least partially, views).

**Comment by HerzScheisse** [comment]
Thanks for your help ric...

But i fear that i dont really understand how this works... i tried to play a bit with the `ManageMediaData.php` but none of my changes were visbile in the media-list page...

can you please assist with some code to get the shared place displayed under the photos`

thanks

**Comment by ric2016** [comment]
Sorry, no, this is out of scope for me personally (and for Vesta).
(I would have to spend time myself getting this to work)
